### DISCUSSION TITLE
@@id([...]) gives warning?

### DISCUSSION THREAD

**Discussion by Coder2195** [prisma/prisma#18409] [Q&A]
### Bug description

I have been using this https://www.prisma.io/docs/concepts/components/prisma-schema/relations/self-relations#many-to-many-self-relations 

All I do is paste in the example 

![image](https://user-images.githubusercontent.com/63069499/226735786-2914731c-72f7-4bf3-890a-144d4538b65a.png)

### How to reproduce

Paste this code into a prisma file
```
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider     = "mysql"
  url          = env("DATABASE_URL")
  relationMode = "prisma"
}

model User {
  id         Int       @id @default(autoincrement())
  name       String?
  followedBy Follows[] @relation("following")
  following  Follows[] @relation("follower")
}

model Follows {
  follower    User @relation("follower", fields: [followerId], references: [id])
  followerId  Int
  followingId Int
  following   User @relation("following", fields: [followingId], references: [id])

  @@id([followerId, followingId])
}
```

### Expected behavior

Expected no warnings since its the boilerplate code provided in the docs

If that is wrong there is something wrong with the docs

### Prisma information

Mentioned above

### Environment & setup

- OS: Ubuntu 22.04
- Database: Planetscale
- Node.js version: 18.14.0

### Prisma Version

```
4.11.0
```

**Comment by janpio** [comment]
The difference is that you are using PlanetScale, which requires `relationMode = "prisma"`, which you have - and that unfortunately leads to different index handling that with other database (and without that relation mode setting) is not a problem. There is not much we can do about it.

**Reply by nurul3101** [reply]
@Coder2195Text Did you had a chance to look at Jan's response?
Let us know in case you still need assistance. It would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏
